src/unit_tests/variant/variant_assign_test.cpp

```diff
@@ -150,6 +150,11 @@ TEST_CASE("move assignment", "[variant]")
 
 TEST_CASE("variant::operator=() - self assignment", "[variant]")
 {
+#if RTTR_COMPILER == RTTR_COMPILER_CLANG || RTTR_COMPILER == RTTR_COMPILER_APPLECLANG
```

Comment:
Does this need implementation for other compilers as well. I think @acki-m  prefers this?